lib/include/tree_sitter/api.h

```diff
@@ -1255,6 +1257,35 @@ TSSymbolType ts_language_symbol_type(const TSLanguage *self, TSSymbol symbol);
  */
 uint32_t ts_language_version(const TSLanguage *self);
 
+/**
+ * Get the ABI version number for this language. This version number is used
+ * to ensure that languages were generated by a compatible version of
+ * Tree-sitter.
+ *
+ * See also [`ts_parser_set_language`].
+ */
+uint32_t ts_language_abi_version(const TSLanguage *self);
+
+/**
+ * Get the [Semantic Version](https://semver.org/) for this language. This
+ * version number is used to signal if a given parser might be incompatible
+ * with existing queries when upgraded between major versions.
+ *
+ * The Semantic Version is encoded as an unsigned 32-bit integer, where the
+ * major, minor, and patch version numbers are encoded in the 24 least significant
+ * bits of the integer, using 8 bits for each number.
+ *
+ * Layout of the returned integer:
+ *
+ * MSB                               LSB
+ * +--------+--------+--------+--------+
+ * |00000000|  Major |  Minor |  Patch |
+ * +--------+--------+--------+--------+
+ * 31      24 23    16 15     8 7      0
+ *
+ */
+uint32_t ts_language_semantic_version(const TSLanguage *self);
```

Comment:
Should we use `uint64_t` to support calendar versioning (`major > UINT8_MAX`)?

Reply:
We could, but then we can't really validate the version at generate time. I guess we should just store a string then, and let downstream figure it out.

Reply:
I'm sorry, but when was calver ever discussed? You set the API, you can mandate semver.

Reply:
Yeah, I, personally, am really for semver.